File: 0049-group-anagrams/0049-group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string,list<string>>m;
        for(auto key : strs)
        {
            string temp = key;
            sort(temp.begin(),temp.end());
            m[temp].push_back(key);
        }
        vector<vector<string>>res;
        for(auto key : m)
        {
            vector<string>temp;
            for(auto key1 : key.second)
            {
                temp.push_back(key1);
            }
            res.push_back(temp);
        }
        return res;
    }
};

File: 0049-group-anagrams/0049-group-anagrams.java
class Solution {
    
    public static String sortString(String s) {
        char[] chars = s.toCharArray();
        Arrays.sort(chars);
        return new String(chars);
    }
    
    public List<List<String>> groupAnagrams(String[] strs) {
        
        
        HashMap<String, ArrayList<String>> map = new HashMap<>();
        
        for(String word : strs){
            
            String sorted = sortString(word);
            ArrayList<String> list= map.getOrDefault(sorted , new ArrayList<>());
            list.add(word);
            map.put(sorted,list);
        }
        
        List<List<String>>res = new ArrayList<>();
        
        for(String word : map.keySet()){
            ArrayList<String>anagrams = map.get(word);
            res.add(anagrams);
        }
        
        return res;

        
    }
}